### ISSUE TITLE
CA2217 creates a false-positive when an enum value is missing

### ISSUE THREAD

**Issue by x3ntrix** [dotnet/roslyn-analyzers#1566]
#### Analyzer package

Microsoft.CodeQuality.Analyzers, version [2.6.0](https://www.nuget.org/packages/Microsoft.CodeQuality.Analyzers/2.6.0)

#### Analyzer

CA2217: Do not mark enums with FlagsAttribute [MSDN](https://msdn.microsoft.com/en-us/library/ms182335.aspx)

#### Repro steps

Create an enumeration with the FlagsAttribute where a value is missing.

```csharp
    [Flags]
    public enum TestEnum
    {
        Monday = 1,
        Wednesday = 4,
        Thursday = 8,
        Friday = 16,
    }
```

#### Expected behavior
Analyzer does not report a warning for the enum. According to MSDN it is sufficient when each value is a power of two, or a combination of defined values. 

#### Actual behavior
Analyzer reports a warning because the value 2 is missing.

**Comment by sharwell** [comment]
:+1:
